Short-lived environments deployed by CD are not destroyed when PR is closed.

### Version

5.31.0

### Operating System

_No response_

### Browser

_No response_

### What are the steps to reproduce this bug?

1. Set up CI/CD
    1. `@webiny/cli-plugin-scaffold-ci` is not work ( reported in #2554, then fixed in #2601 ). Therefore, copy files from `packages/cli-plugin-scaffold-ci/src/githubActions/files/workflows/.github` in this repository to `.github` in your project repository. 
    2. For other settings, refer to [the document](https://www.webiny.com/docs/core-development-concepts/ci-cd/setup).
        1. Set “passphrase” as DEV_PULUMI_SECRETS_PROVIDER, and the S3 URI as DEV_WEBINY_PULUMI_BACKEND.
2. Create PR
3. Merge PR, then `Pull Requests (Closed)` workflow is started, but stopped with the error ( `webiny error: Command failed with ENOENT: /home/runner/work/YOUR_REPOSITORY_NAME/YOUR_REPOSITORY_NAME/.webiny/pulumi-cli/linux/pulumi/pulumi login ***/apps/website --non-interactive` ).

### What is the expected behavior?

Short-lived environments deployed on AWS are destroyed without errors.

### What do you see instead?

```
webiny error: Command failed with ENOENT: /home/runner/work/YOUR_REPOSITORY_NAME/YOUR_REPOSITORY_NAME/.webiny/pulumi-cli/linux/pulumi/pulumi login ***/apps/website --non-interactive
spawn /home/runner/work/YOUR_REPOSITORY_NAME/YOUR_REPOSITORY_NAME/.webiny/pulumi-cli/linux/pulumi/pulumi ENOENT

Error: Process completed with exit code 1.
##[debug]Finishing: Destroy Website project application
```

### Additional information

_No response_

### Possible solution

This `ENOENT` can be resolved by creating the following directory.

```
.webiny/workspaces/apps/website
.webiny/workspaces/apps/admin
.webiny/workspaces/apps/api
.webiny/workspaces/apps/core
```

But when I do, I still get the error below and the short-lived environment is not destroyed.

```
webiny error: Project application apps/website (pr#ISSUE_NUBER environment) does not exist.
```

Those errors can be resolved by editing the `.github/workflows/pullRequestClosed.yml` as follows.

```.diff
diff --git a/.github/workflows/pullRequestClosed.yml b/.github/workflows/pullRequestClosed.yml
index 483023a..f773397 100644
--- a/.github/workflows/pullRequestClosed.yml
+++ b/.github/workflows/pullRequestClosed.yml
@@ -53,13 +53,53 @@ jobs:
 
       # 2. Destroy short-lived environment.
       - name: Destroy Website project application
-        run: yarn webiny destroy apps/website --env pr${{ github.event.pull_request.number }}
+        run: |
+          mkdir -p .webiny/workspaces/apps/website
+          if [[ ${{ secrets.DEV_WEBINY_PULUMI_BACKEND }} =~ ^s3://.* ]]; then
+            cat <<EOF > Pulumi.yaml
+          name: website
+          runtime: nodejs
+          backend:
+            url: ${{ secrets.DEV_WEBINY_PULUMI_BACKEND }}/apps/website
+          EOF
+          fi
+          yarn webiny destroy apps/website --env pr${{ github.event.pull_request.number }}
 
       - name: Destroy Admin Area project application
-        run: yarn webiny destroy apps/admin --env pr${{ github.event.pull_request.number }}
+        run: |
+          mkdir -p .webiny/workspaces/apps/admin
+          if [[ ${{ secrets.DEV_WEBINY_PULUMI_BACKEND }} =~ ^s3://.* ]]; then
+            cat <<EOF > Pulumi.yaml
+          name: admin
+          runtime: nodejs
+          backend:
+            url: ${{ secrets.DEV_WEBINY_PULUMI_BACKEND }}/apps/admin
+          EOF
+          fi
+          yarn webiny destroy apps/admin --env pr${{ github.event.pull_request.number }}
 
       - name: Destroy API project application
-        run: yarn webiny destroy apps/api --env pr${{ github.event.pull_request.number }}
+        run: |
+          mkdir -p .webiny/workspaces/apps/api
+          if [[ ${{ secrets.DEV_WEBINY_PULUMI_BACKEND }} =~ ^s3://.* ]]; then
+            cat <<EOF > Pulumi.yaml
+          name: api
+          runtime: nodejs
+          backend:
+            url: ${{ secrets.DEV_WEBINY_PULUMI_BACKEND }}/apps/api
+          EOF
+          fi
+          yarn webiny destroy apps/api --env pr${{ github.event.pull_request.number }}
 
       - name: Destroy Core project application
-        run: yarn webiny destroy apps/core --env pr${{ github.event.pull_request.number }}
+        run: |
+          mkdir -p .webiny/workspaces/apps/core
+          if [[ ${{ secrets.DEV_WEBINY_PULUMI_BACKEND }} =~ ^s3://.* ]]; then
+            cat <<EOF > Pulumi.yaml
+          name: core
+          runtime: nodejs
+          backend:
+            url: ${{ secrets.DEV_WEBINY_PULUMI_BACKEND }}/apps/core
+          EOF
+          fi
+          yarn webiny destroy apps/core --env pr${{ github.event.pull_request.number }}
```

↓test result
<img width="1030" alt="スクリーンショット 2022-09-06 1 43 34" src="https://user-images.githubusercontent.com/2719533/188491268-489d5805-d9ba-4abe-ba5b-823c4d3e8cca.png">

If you agree with this change, I will create a PR with this change to [pullRequestClosed.yml](https://github.com/webiny/webiny-js/blob/next/packages/cli-plugin-scaffold-ci/src/githubActions/files/workflows/.github/workflows/pullRequestClosed.yml).

--- a/packages/cli-plugin-deploy-pulumi/commands/destroy.js
+++ b/packages/cli-plugin-deploy-pulumi/commands/destroy.js
@@ -2,7 +2,9 @@ const { createPulumiCommand, processHooks } = require("../utils");
 
 module.exports = createPulumiCommand({
     name: "destroy",
-    createProjectApplicationWorkspace: false,
+    // We want to create a workspace just because there are cases where the destroy command is called
+    // without the deployment happening initially (e.g. CI/CD scaffold's `pullRequestClosed.yml` workflow).
+    createProjectApplicationWorkspace: true,
     command: async ({ inputs, context, projectApplication, pulumi, getDuration }) => {
         const { env, folder } = inputs;
 

